components/org.wso2.carbon.identity.oauth.ciba/src/main/java/org/wso2/carbon/identity/oauth/ciba/dao/CibaMgtDAOImpl.java

```diff
@@ -290,24 +294,27 @@ public CibaAuthCodeDO getCibaAuthCode(String authCodeKey) throws CibaCoreExcepti
                     CibaSQLQueries.RETRIEVE_AUTH_CODE)) {
 
                 prepStmt.setString(1, authCodeKey);
-                ResultSet resultSet = prepStmt.executeQuery();
-                if (resultSet.next()) {
-                    cibaAuthCodeDO.setCibaAuthCodeKey(resultSet.getString(1));
-                    cibaAuthCodeDO.setAuthReqID(resultSet.getString(2));
-                    cibaAuthCodeDO.setConsumerKey(resultSet.getString(3));
-                    cibaAuthCodeDO.setLastPolledTime(
-                            resultSet.getTimestamp(4, Calendar.getInstance(TimeZone.getTimeZone(CibaConstants.UTC))));
-                    cibaAuthCodeDO.setInterval(resultSet.getLong(5));
-                    cibaAuthCodeDO.setExpiresIn(resultSet.getLong(6));
-                    cibaAuthCodeDO.setAuthReqStatus(AuthReqStatus.valueOf(resultSet.getString(7)));
-                    cibaAuthCodeDO.setIssuedTime(
-                            resultSet.getTimestamp(8, Calendar.getInstance(TimeZone.getTimeZone(CibaConstants.UTC))));
-                } else {
-                    return null;
-                }
+                try (ResultSet resultSet = prepStmt.executeQuery()) {
+                    if (resultSet.next()) {
+                        cibaAuthCodeDO.setCibaAuthCodeKey(resultSet.getString(1));
+                        cibaAuthCodeDO.setAuthReqId(resultSet.getString(2));
+                        cibaAuthCodeDO.setConsumerKey(resultSet.getString(3));
+                        cibaAuthCodeDO.setLastPolledTime(
+                                resultSet.getTimestamp(4,
+                                        Calendar.getInstance(TimeZone.getTimeZone(CibaConstants.UTC))));
+                        cibaAuthCodeDO.setInterval(resultSet.getLong(5));
+                        cibaAuthCodeDO.setExpiresIn(resultSet.getLong(6));
+                        cibaAuthCodeDO.setAuthReqStatus(AuthReqStatus.valueOf(resultSet.getString(7)));
+                        cibaAuthCodeDO.setIssuedTime(
+                                resultSet.getTimestamp(8,
+                                        Calendar.getInstance(TimeZone.getTimeZone(CibaConstants.UTC))));
+                    } else {
+                        return null;
```

Comment:
Wondering whether we should throw an exception here instead of returning null. Wdyt?

Reply:
No need to throw an error here. If there is no record we can expect to return null and handle accordingly ?